Base/Crypto/Crypto.Static/RSAKeyManager.hpp

```diff
@@ -0,0 +1,210 @@
+            _keyPair.privateKey.BERDecode(privateKeySource);
+            privateKeyLoaded = true;
+        }
+        privateKeyOutputFile.close();
+
+        if (privateKeyLoaded)
+        {
+            if (AutoSeededRandomPool randPool; _keyPair.privateKey.Validate(randPool, ValidationLevel::HARD))
+            {
+                FileLogger::getInstance().log("[SERVER] Private key loaded from file successfully",
+                                                            Base::Logger::LogLevel::INFO);
+                loadPublicKey();
+            }
+            else
+            {
+                FileLogger::getInstance().log("[SERVER] Invalid (probably compromised) private key", Base::Logger::LogLevel::WARNING);
+                _keyPair = RSAKeyGenerator().generateRSAKeyPair();
+            }
+        }
+        else
+        {
+            FileLogger::getInstance().log("[SERVER] Private key not found", Base::Logger::LogLevel::INFO);
+            _keyPair = RSAKeyGenerator().generateRSAKeyPair();
+        }
+    }
+
+     /** @brief Method for loading public key from a file
+     * @details Method loads public RSA key from file "PUBLIC_KEY_FILE". The following situations are possible: \
+     * - Public key not loaded  -> generate a public key based on the uploaded private key  \
+     * - File not found -> generate a public key based on the uploaded private key  \
+     * - Public key loaded and valid -> end of key loading
+     */
+    inline void loadPublicKey()
+    {
+        bool publicKeyLoaded = false;
+
+        std::ifstream publicKeyOutputFile(PUBLIC_KEY_FILE);
+        if (publicKeyOutputFile.is_open())
+        {
+            publicKeyOutputFile.close();
+
+            FileSource publicKeySource(PUBLIC_KEY_FILE.c_str(), true);
+            _keyPair.publicKey.BERDecode(publicKeySource);
+            publicKeyLoaded = true;
+        }
+
+        if (AutoSeededRandomPool randPool;
+            !publicKeyLoaded || !_keyPair.publicKey.Validate(randPool, ValidationLevel::BALANCED))
+        {
+            FileLogger::getInstance().log("[SERVER] Public key not found or invalid", Base::Logger::LogLevel::INFO);
+            _keyPair.publicKey = RSAKeyGenerator().generateRSAPublicKey(_keyPair.privateKey);
+        }
+        else
+        {
+            FileLogger::getInstance().log("[SERVER] Public key loaded from file successfully", Base::Logger::LogLevel::INFO);
+        }
+        publicKeyOutputFile.close();
+    }
+
+    /** @brief Method for saving private RSA key in file "PRIVATE_KEY_FILE"
+    * @details Saves the private key by completely overwriting the file. Failed private key saving does not
+    * startup public key saving.
+    */
+    GeneralCodes savePrivateKey()
+    {
+        std::ofstream privateKeyInputFile(PRIVATE_KEY_FILE);
+        if (privateKeyInputFile.is_open())
+        {
+            privateKeyInputFile.close();
+
+            FileSink privateKeySink(PRIVATE_KEY_FILE.c_str() /*ios::trunc*/);
```

Comment:
Useless comm or not?

Reply:
This is constructor from CryptoPP library and file opening mode is hidden for you, so this comment makes sense.